addons/website/static/src/components/views/page_views_mixin.js

```diff
@@ -31,7 +31,10 @@ export const PageControllerMixin = (component) => class extends component {
         onWillStart(async () => {
             await this.website.fetchWebsites();
             this.websiteSelection.push(...this.website.websites);
-            this.state.activeWebsite = this.website.currentWebsite || this.website.websites[0];
+            const currentWebsite = (await this.orm.call('website', 'get_current_website')).match(/\d+/);
+            this.state.activeWebsite = currentWebsite ?
+                this.website.websites.find(w => w.id === parseInt(currentWebsite[0])) :
+                this.website.websites[0];
```

Comment:
Nitpicking I guess but is this possible to not have a website returned by the ORM? It always falls back to the first website which cannot have been deleted no? (and if it was deleted somehow, the user has far more problems than trying to access the page manager ^^).
```suggestion
            const res = await this.orm.call('website', 'get_current_website');
            const currentWebsite = parseInt(res.match(/d+/)[0]);
            this.state.activeWebsite = this.website.websites.find(w => w.id === parseInt(currentWebsite[0]));
```

Reply:
No, we can't, as it's not guaranteed that a website_id will be returned by get_current_website (regardless of the `fallback=True` param, so the docstring is not 100% accurate):
https://github.com/odoo/odoo/blob/533e333b62386fbd2587468634955e91827e8787/addons/website/models/website.py#L949-L958

I actually wrote it the kind of same way you did at first:
```js
const websiteId = (await this.orm.call('website', 'get_current_website')).match(/\d+/)[0];
```
But there was cases of empty record `website()` being returned, making this kind of code crash.

Reply:
> But there was cases of empty record website() being returned

What were those cases? Are not they bugs on their own? That what you wrote actually revealed? From what I see from the implementation, the only possibility is having a forced website_id in the context (in case of session, we actually clean it) but that website does not exist... is not that a bug on its own? :open_mouth: 

But ok not to change in stable if you prefer not to.

Reply:
`get_current_website` was problematic in some cases historically, it either returned a website when it should not, or did not return a website when it should, I don't remember exactly. https://github.com/odoo/odoo/commit/2391d0994ae564347ac01772811f5cfc22e951ff should have improved that tho.

It's also possible if you set `fallback=False`, which is not the case here.

> What were those cases? Are not they bugs on their own? That what you wrote actually revealed?

I honestly have no clue, I don't remember why it failed, I just fixed it directly as it failed the first time I ran this line :grimacing: 
But clearly, I was not playing with deleted website from the context. I played with so many flows that it's hard to remember: multi website + multi domain (with and without one no-domain website) etc.
It was maybe also failing only in Odoo 16.4., I gave it a quick look to make it fail after the retarget to 16 and it seems like I can't make it return an empty website.

As it's the page manager in Odoo 16, and since I faced this issue without doing anything too crazy, I am not sure about removing that safety check tho.

AFAIC, I'd add a comment to mention it should not be possible but it's a safety net?

While waiting for the green runbot, I'll reapply the commit on 16.4 and give it a shot to see if I can make it fail there again

Reply:
Or just let's say I broke something after testing so many flows, maybe a non-legit way, and ship it without that safety net.
If it fails, we should know it quickly. I tried a few flows and nothing broke.

Reply:
> AFAIC, I'd add a comment to mention it should not be possible but it's a safety net?

IMO no need of a comment (but you can add one, it's never too much I guess): I would merge with the safety net in stable and remove it in the master version. IMO safety nets "hide" more bugs than they prevents, so I would rather let master crash if the case actually exist so we can investigate (of course if we agree that `get_current_website` not returning a website when `fallback=True` is not a desired behavior). Of course, in stable, if you saw a bug without the safety net, of course we should not remove it. If you prefer to keep it in master as well, as you wish, I'll forget anyway since you'll be merging the forward-port :+1:

Reply:
Meh, replied "at the same time" as you. Confirm if you don't want to change your mind :+1: (Tell me once ready to be merged again :+1:)

Reply:
It's back :D
https://github.com/odoo/odoo/compare/52851659a1fe9db9c17c6a940a6f589992dedd2b..9df3529a35f33055dcff3dc4775ee5b09cfaaa10
Should be ready, hopefully the runbot should be green.
Everything runs fine in local.

Note that I added a test for the point 2. of the commit message (point 2. = the one which made it so we re-targeted the PR to 16), as you mentioned https://github.com/odoo/odoo/pull/149070#pullrequestreview-1816122022